var/spack/repos/builtin/packages/amg2023/package.py

```diff
@@ -18,66 +18,34 @@ class Amg2023(MakefilePackage):
     git = "https://github.com/LLNL/AMG2023.git"
 
     version("develop", branch="main")
```

Comment:
Since main doesn't have this support, building `amg2023@develop` would break. Normally when there are two build systems, you can add a `build_system` variant (the `fastani` package is an example of this). Since my understanding is that make support will be removed entirely, and because the only other version in here is a moving target (i.e. the tip of develop), I don't consider that an issue, but FYI.